gapic-generator-java/src/main/java/com/google/api/generator/engine/writer/JavaWriterVisitor.java

```diff
@@ -1034,6 +1034,9 @@ public void visit(ClassDefinition classDefinition) {
 
   @Override
   public void visit(PackageInfoDefinition packageInfoDefinition) {
+    if (packageInfoDefinition.shouldGenerateFile()) {
+      return;
+    }
```

Comment:
Same thing: `!packageInfoDefinition.shouldGenerateFile()`? If we shouldn't generate, then we exit quickly?

Reply:
The purpose of this change was to avoid producing a srcjar because this function will return a non-empty string even if we have no services. I now changed this to simply no-op as `Writer` will exit early and not create a file when no services are found.